src/tools/orb_radar.rs

```diff
@@ -269,29 +269,27 @@ impl OrbRadar {
             painter.arrow(
                 circle_pos - arrow / 2.0,
                 arrow,
-                Stroke::new(stroke.width, self.orb_color(ui, first_orb)),
+                Stroke::new(stroke.width, self.orb_color(ui, first_orb, state)),
             );
 
             painter.text(
                 circle_pos + vec2(radius + padding, 0.0),
                 Align2::LEFT_CENTER,
                 format!("{dist_to_first:.1} px"),
                 text_font,
-                self.orb_color(ui, first_orb),
+                self.orb_color(ui, first_orb, state),
             );
         });
     }
 
-    fn orb_color(&self, ui: &mut Ui, orb: &Orb) -> Color32 {
-        match orb.source {
-            OrbSource::Room => {
-                if !orb.corrupted {
-                    ui.style().visuals.hyperlink_color
-                } else {
-                    ui.style().visuals.error_fg_color
-                }
+    fn orb_color(&self, ui: &Ui, orb: &Orb, state: &AppState) -> Color32 {
+        if !self.show_rooms {
+            ui.style().visuals.text_color()
+        } else {
+            match orb.source {
```

Comment:
nit: I am a huge mega turbo ultra fan of early returns, especially(!) if the else is larger than the if

Reply:
Done